File: Ruby/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaJuez < Casilla
    def initialize (carcel, nombre)
      @carcel = carcel
      super(nombre)
    end
    
    def recibe_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        super
        todos[actual].encarcelar(@carcel)
      end
    end
    
    def to_s
      return super + "\nJugador enviado a la cárcel\n"
    end
  end
end

File: Java/CasillaJuez.java
package civitas;

import java.util.ArrayList;

public class CasillaJuez extends Casilla {

    private static int carcel;

    public CasillaJuez(int _carcel, String nombre) {
        super(nombre);
        carcel = _carcel;
    }

    @Override
    public String toString() {
        return super.toString() + "\nJugador enviado a la cárcel\n";
    }

    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (super.jugadorCorrecto(actual, todos)) {
            super.informe(actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
}
